pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -58,6 +59,34 @@ type ClientCache struct {
 	// argument evaluation
 	discoveryClientFactory DiscoveryClientFactory
 	discoveryClient        discovery.DiscoveryInterface
+
+	kubernetesClientLoader KubernetesClientCache
+}
+
+// kubernetesClientLoader provides a new kubernetes.Clientset
+type KubernetesClientCache struct {
+	// once makes sure the client is only initialized once
+	once sync.Once
+	// client is the cached client value
+	client *kubernetes.Clientset
+	// err is the cached error value
+	err error
+}
+
+// KubernetesClientSet returns a new kubernetes.Clientset.  It will cache the value
+// the first time it is called and return the cached value on subsequent calls.
+// If an error is encountered the first time KubernetesClientSet is called,
+// the error will be cached.
+func (c *ClientCache) KubernetesClientSet(requiredVersion *schema.GroupVersion) (*kubernetes.Clientset, error) {
```

Comment:
Trying to keep the naming consistent with the sister function that works on internal clientsets - `ClientSetForVersion`.  Renamed to `KubernetesClientSetForVersion`